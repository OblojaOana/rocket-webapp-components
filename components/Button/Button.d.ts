import { ButtonProps as BaseProps } from '@material-ui/core/Button';

export interface ButtonProps extends Omit<BaseProps, 'color' | 'size'> {
  /**
  * The color of the button.
  */
  color?: "primary" |
  "info" |
  "theme" |
  "themeNoBackground" |
  "themeWithBackground" |
  "success" |
  "warning" |
  "danger" |
  "rose" |
  "white" |
  "simple" |
  "defaultNoBackground" |
  "primaryNoBackground" |
  "infoNoBackground" |
  "successNoBackground" |
  "warningNoBackground" |
  "dangerNoBackground" |
  "roseNoBackground" |
  "white" |
  "simple" |
  "transparent"
  /**
   * If true, rounded corners are enabled.
   */
  round?: boolean
  /**
  * 	The content of the component.
  */
  children?: React.ReactNode
  /**
  * 	If true, the button will take up the full width of its container.
  */
  fullWidth?: boolean
  /**
  * If true, the button will be disabled.
  */
  disabled?: boolean
  /**
  * A custom class provided.
  */
  customClass?: string
  /**
  * If true,  the button's min width will be set to 160px.
  */
  wd?: boolean
  /**
  * If true, the button will be smaller.
  */
  justIcon?: boolean
  /**
  * If true, the button will float to the right.
  */
  right?: boolean
  /**
  * The size of the button.
  */
  size?: "sm" | "lg" | "xs"
  /**
  * The tooltip of the button.
  */
  tooltip?: string

}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/button
 *
 *   Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/Button/Button.js
 */
export default function Button(props: ButtonProps): JSX.Element;