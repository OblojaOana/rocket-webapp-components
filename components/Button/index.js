import Button from "../../core/Button";
export default Button;