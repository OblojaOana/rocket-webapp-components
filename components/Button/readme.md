##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import Button from '@bit/totalsoft_oss.react-mui.button';
import { makeStyles } from '@material-ui/core/styles';
import DeleteIcon from '@material-ui/icons/Delete';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import KeyboardVoiceIcon from '@material-ui/icons/KeyboardVoice';
import Icon from '@material-ui/core/Icon';
import SaveIcon from '@material-ui/icons/Save';
import SendIcon from '@material-ui/icons/Send';

const useStyles = makeStyles((theme) => ({
	button: {
		margin: theme.spacing(5),
	},
}));

const App = () => {
	const classes = useStyles();

	return (
		<div>
			<Button
				color="warning"
				className={classes.button}
				startIcon={<DeleteIcon />}
				size="sm"
			>
				Delete
      </Button>
			<Button
				color="primary"
				className={classes.button}
				endIcon={<SendIcon />}
				size="sm"
			>
				Send
      </Button>
			<Button
				color="info"
				className={classes.button}
				startIcon={<CloudUploadIcon />}
				size="sm"
			>
				Upload
      </Button>
			<Button
				color="danger"
				className={classes.button}
				startIcon={<KeyboardVoiceIcon />}
				size="sm"
			>
				Talk
      </Button>
			<Button
				color="info"
				size="small"
				className={classes.button}
				startIcon={<SaveIcon />}
				size="sm"
			>
				Save
      </Button>
			<Button
				color="success"
				size="large"
				className={classes.button}
				startIcon={<SaveIcon />}
				size="sm"
			>
				Save
      </Button>
		</div>
	)
}

export default (
	<App />
)
 ```



 ##### Copy this snapshot in the overview panel
 ```js
import React from 'react';
import Button from '@bit/totalsoft_oss.react-mui.button';
import { makeStyles } from '@material-ui/core/styles';
import DeleteIcon from '@material-ui/icons/Delete';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';
import KeyboardVoiceIcon from '@material-ui/icons/KeyboardVoice';
import Icon from '@material-ui/core/Icon';
import SaveIcon from '@material-ui/icons/Save';
import SendIcon from '@material-ui/icons/Send';

const useStyles = makeStyles((theme) => ({
	margin: {
		margin: theme.spacing(1),
	},
	extendedIcon: {
		marginRight: theme.spacing(1),
	},
}));

const App = () => {
	const classes = useStyles();

	return (
		<div>
			<Button size="xs" color="info" className={classes.margin}>
				Small
        </Button>
			<Button size="sm" color="info" className={classes.margin}>
				Medium
        </Button>
			<Button size="lg" color="info" className={classes.margin}>
				Large
        </Button>
		</div>
	)
}

export default (
	<App />
)
 ```