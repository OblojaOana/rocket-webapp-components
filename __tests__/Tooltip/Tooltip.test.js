import React from "react";
import renderer from "react-test-renderer";
import Tooltip from "../../components/Tooltip/Tooltip";
import Button from "../../components/Button/Button";

describe("Tooltip", () => {
  it("Snapshot", () => {
    const component = renderer.create(
      <Tooltip title="Tooltip">
        <Button>Hover on me</Button>
      </Tooltip>
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
