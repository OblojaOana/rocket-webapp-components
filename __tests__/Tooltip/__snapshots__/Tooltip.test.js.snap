// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Tooltip"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1 "
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Hover on me
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</span>
`;
